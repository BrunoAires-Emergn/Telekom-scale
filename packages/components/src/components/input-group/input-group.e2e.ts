import { newE2EPage } from '@stencil/core/testing';

describe('scale-input-group', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-input-group/>');
    const element = await page.find('scale-input-group');
    expect(element).toHaveClass('hydrated');
  });
});
