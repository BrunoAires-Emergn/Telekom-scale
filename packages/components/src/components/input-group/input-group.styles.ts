import { JssStyle } from 'jss';

export const styles: JssStyle = {
  'input-group': {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    marginBottom: '1rem',
  },
};
