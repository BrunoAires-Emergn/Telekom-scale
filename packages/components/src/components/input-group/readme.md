# t-input-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type                                     | Default     |
| ------------- | -------------- | ------------------------------ | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Input text class    | `string`                                 | `''`        |
| `styles`      | --             | (optional) Injected jss styles | `StyleSheet<string \| number \| symbol>` | `undefined` |


## Methods

### `updateStyles(newStyle: StyleSheet<string | number | symbol>) => Promise<void>`

Method: updateStyles()

#### Returns

Type: `Promise<void>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
