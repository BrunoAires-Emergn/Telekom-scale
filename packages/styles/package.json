{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
