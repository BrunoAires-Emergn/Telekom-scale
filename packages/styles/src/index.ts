import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
