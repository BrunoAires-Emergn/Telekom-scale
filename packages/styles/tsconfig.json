{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
