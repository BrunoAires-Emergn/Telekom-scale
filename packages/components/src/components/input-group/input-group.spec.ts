import { newSpecPage } from '@stencil/core/testing';
import { InputGroup } from './input-group';
import { styles } from './input-group.styles';
import jss from 'jss';

describe('Input Group', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new InputGroup();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [InputGroup],
      html: `<t-input-group>Label</t-input-group>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // it('should handle a custom css class', () => {
  //   element.customClass = 'custom';
  //   expect(element.getCssClassMap()).toContain('custom');
  // });

  // it('should handle theme css class', () => {
  //   element.theme = 'default';
  //   expect(element.getCssClassMap()).toContain('input-group--theme-default');
  // });

  // it('should have a default css class', () => {
  //   expect(element.getCssClassMap()).toContain('input-group');
  // });
});
