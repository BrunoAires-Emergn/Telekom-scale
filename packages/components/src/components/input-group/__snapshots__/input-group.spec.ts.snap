// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Group should match snapshot 1`] = `
<t-input-group>
  <mock:shadow-root>
    <style>
      .input-group-0-0-2 {
  width: 100%;
  display: flex;
  margin-bottom: 1rem;
  flex-direction: column;
}
    </style>
    <div class="input-group-0-0-2">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Label
</t-input-group>
`;
